src/models/step_arguments/data_table_spec.js

```diff
@@ -44,6 +44,141 @@ describe('DataTable', function () {
     })
   })
 
+  describe('typed hashes', function() {
+    it('rejects tables where not all rows have 3 columns', function() {
+      const dataTable = new DataTable({
+        rows: [
+          {
+            cells: [
+              {value: 'key1'},
+              {value: 'string'},
+              {value:  'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key2'},
+              {value: 'string'},
+              {value: 'value'}
+            ]
+          } , {
+            cells: [
+              {value: 'key3'},
+              {value: 'value'}
+            ]
+          }
+        ]
+      })
+      expect(() => {
+        dataTable.typedRowsHash()
+      }).to.throw(Error)
```

Comment:
I actually remember thinking about it already back then, but wasn't sure it was necessary
Thanks for insisting!